src/ray.c

```diff
@@ -26,14 +25,15 @@ t_colour	ft_ray_colour(t_ray ray, t_entities scene)
 	t_interval	ray_d;
 	int			i;
 
-	ray_d.min = 0;
-	ray_d.max = INFINITY;
-	rec.d = INFINITY;
+	ft_init_colour(&ray_colour);
+	ft_init_hitrecord(&rec);
+	rec.ray = ray;
+	ft_init_interval(&ray_d);
```

Comment:
No, as the interval max value needs to be adapted further on to exclude all hittables which are farther away than the current hit. 
if (rec.d < ray_d.max)
	ray_d.max = rec.d;

Reply:
Then assignment of (t_interval){.min = 0, .max = 0} would suffice.

Argument against this way of init function: the variable ray_d lies on the stack (and dies after this function call). We need to pass the address of it to this init function. This means the compiler can't really optimize, he doesn't know the address beforehand. This makes it slow.
If the init function would return an initializes t_interval it could be optimized.
This of course begs the question why even make it a function if we can just assign it compound literal it.

Reply:
Replaced with compound literal